ast/expressions.go

```diff
@@ -269,7 +269,10 @@ type CaseExpr struct {
 // Format the ExprNode into a Writer.
 func (n *CaseExpr) Format(w io.Writer) {
 	fmt.Fprint(w, "CASE ")
-	n.Value.Format(w)
+	// Because the presence of case when syntax, Value could be nil and we need check this.
```

Comment:
Can not understand the comment.

Reply:
because `case when`, `CaseExpr`'s `Value` is nil. The original code fails to consider this that why we meet a nil pointer reference. This checker is to add to avoid this situation.